packages/flutter_tools/lib/src/test/coverage_collector.dart

```diff
@@ -141,3 +154,120 @@ class CoverageCollector extends TestWatcher {
+  await service.getVM();
+  final List<Map<String, dynamic>> coverage = <Map<String, dynamic>>[];
+
+  for (Isolate isolateRef in service.vm.isolates) {
+    await isolateRef.load();
+    final Map<String, dynamic> scriptList = await isolateRef.invokeRpcRaw('getScripts', params: <String, dynamic>{'isolateId': isolateRef.id});
+    final List<Future<void>> futures = <Future<void>>[];
+
+    final Map<String, Map<String, dynamic>> scripts = <String, Map<String, dynamic>>{};
+    final Map<String, Map<String, dynamic>> sourceReports = <String, Map<String, dynamic>>{};
+    for (Map<String, dynamic> script in scriptList['scripts']) {
+      if (!libraryPredicate(script['uri'])) {
+        continue;
+      }
+      final String scriptId = script['id'];
+      futures.add(
+        isolateRef.invokeRpcRaw('getSourceReport', params: <String, dynamic>{'forceCompile': true, 'scriptId': scriptId, 'isolateId': isolateRef.id, 'reports': <String>['Coverage']})
+          .then((Map<String, dynamic> report) {
+            sourceReports[scriptId] = report;
+          })
+      );
+      futures.add(
+        isolateRef.invokeRpcRaw('getObject', params: <String, dynamic>{'isolateId': isolateRef.id, 'objectId': scriptId})
+          .then((Map<String, dynamic> script) {
+            scripts[scriptId] = script;
+          })
+      );
+    }
+    await Future.wait(futures);
+    final Map<Uri, Map<int, int>> hitMaps = <Uri, Map<int, int>>{};
+    for (String scriptId in scripts.keys) {
+      final Map<String, dynamic> sourceReport = sourceReports[scriptId];
+      for (Map<String, dynamic> range in sourceReport['ranges']) {
+        final Map<String, dynamic> coverage = range['coverage'];
+        final Map<String, dynamic> scriptRef = sourceReport['scripts'][range['scriptIndex']];
+        final Uri uri = Uri.parse(scriptRef['uri']);
+        hitMaps[uri] ??= <int, int>{};
+        final Map<int, int> hitMap = hitMaps[uri];
+        final List<dynamic>  hits = coverage['hits'];
+        final List<dynamic> misses = coverage['misses'];
+        final List<dynamic> tokenPositions = scripts[scriptRef['id']]['tokenPosTable'];
+        if (hits != null) {
+          for (dynamic hit in hits) {
+            final int line = _lineAndColumn(hit, tokenPositions)[0];
+            final int current = hitMap[line] ?? 0;
+            hitMap[line] = current + 1;
+          }
+        }
+        if (misses != null) {
+          for (dynamic miss in misses) {
+            final int line = _lineAndColumn(miss, tokenPositions)[0];
+            hitMap[line] ??= 0;
+          }
+        }
+      }
+    }
+    hitMaps.forEach((Uri uri, Map<int, int> hitMap) {
+      coverage.add(_toScriptCoverageJson(uri, hitMap));
+    });
+  }
+  return <String, dynamic>{'type': 'CodeCoverage', 'coverage': coverage};
+}
+
+List<int> _lineAndColumn(int position, List<dynamic> tokenPositions) {
```

Comment:
This method could use some comments.

Reply:
Added a description and a link to the vmservice documentation which describes the format